lib/unicorn_wrangler.rb

```diff
@@ -25,6 +26,21 @@ def setup(
       @handlers << RequestKiller.new(logger, stats, kill_after_requests) if kill_after_requests
       @handlers << OutOfMemoryKiller.new(logger, stats, kill_on_too_much_memory) if kill_on_too_much_memory
       @handlers << OutOfBandGC.new(logger, stats, gc_after_request_time) if gc_after_request_time
+
+      if map_term_to_quit
+        # - heroku docs say to also set after_fork, but looking into unicorns source shows that
+        #   all traps get unset after_fork already.
```

Comment:
As far as I understand what Heroku suggest is not to unset a trap but to trap `TERM` and do nothing. So that all children are totally immune to `TERM` and will just wait for the master to send QUIT

```ruby
after_fork do |server, worker|
  Signal.trap 'TERM' do
    puts 'Unicorn worker intercepting TERM and doing nothing. Wait for master to send QUIT'
  end
  # ...
end

```

Reply:
tried that, but it never get's called ... I think their code was old and from before unicorn unset all traps

Reply:
sending TERM directly to the child will trigger the after-fork, but nobody should be doing this ... I'd prefer not to add code that I don't understand :)

Reply:
ahhh in heroku it sends the signal to all processes ...

Reply:
Even if the general idea in the interwebs is `A SIGTERM signal is sent to the main process (PID 1) in each container`